### PR TITLE
402 userprofile shouted tab

### PR DESCRIPTION
**Pull request by Tirokk**
> [ogerly](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-05-24T10:50:04Z" title="Friday, May 24th 2019, 12:50:04 pm +02:00">May 24, 2019</time>_
_Closed <time datetime="2019-06-05T08:58:26Z" title="Wednesday, June 5th 2019, 10:58:26 am +02:00">Jun 5, 2019</time>_
---

## 🍰 Pullrequest
402-userprofile-shouted-tab - text adapted for single or multiple. / css extended.
 
gif

### Issues
- relates 402

### Todo

- [X] None

### CODE REVIEW

**Comment by Tirokk** [comment]
> [codecov[bot]](https://github.com/apps/codecov) **Authored by [codecov[bot]](https://github.com/apps/codecov)**
_<time datetime="2019-05-24T13:27:05Z" title="Friday, May 24th 2019, 3:27:05 pm +02:00">May 24, 2019</time>_

---


# [Codecov](https://codecov.io/gh/Human-Connection/Human-Connection/pull/672?src=pr&el=h1) Report
> Merging [#672](https://codecov.io/gh/Human-Connection/Human-Connection/pull/672?src=pr&el=desc) into [master](https://codecov.io/gh/Human-Connection/Human-Connection/commit/9d1dfe6067280b792b752b46608e443aa3925616?src=pr&el=desc) will **decrease** coverage by `0.08%`.
> The diff coverage is `0%`.

[Impacted file tree graph](https://codecov.io/gh/Human-Connection/Human-Connection/pull/672?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master     #672      +/-   ##
==========================================
- Coverage    20.2%   20.12%   -0.08%     
==========================================
  Files         157      157              
  Lines        2208     2212       +4     
  Branches      105      105              
==========================================
- Hits          446      445       -1     
- Misses       1738     1743       +5     
  Partials       24       24
```


| [Impacted Files](https://codecov.io/gh/Human-Connection/Human-Connection/pull/672?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [webapp/pages/profile/\_id/\_slug.vue](https://codecov.io/gh/Human-Connection/Human-Connection/pull/672/diff?src=pr&el=tree#diff-d2ViYXBwL3BhZ2VzL3Byb2ZpbGUvX2lkL19zbHVnLnZ1ZQ==) | `0% <0%> (ø)` | :arrow_up: |
| [webapp/pages/settings/my-social-media.vue](https://codecov.io/gh/Human-Connection/Human-Connection/pull/672/diff?src=pr&el=tree#diff-d2ViYXBwL3BhZ2VzL3NldHRpbmdzL215LXNvY2lhbC1tZWRpYS52dWU=) | `61.9% <0%> (-1.73%)` | :arrow_down: |
| [backend/src/middleware/activityPubMiddleware.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/672/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvbWlkZGxld2FyZS9hY3Rpdml0eVB1Yk1pZGRsZXdhcmUuanM=) | `0% <0%> (ø)` | :arrow_up: |
| [backend/src/activitypub/routes/serveUser.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/672/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvYWN0aXZpdHlwdWIvcm91dGVzL3NlcnZlVXNlci5qcw==) | `0% <0%> (ø)` | :arrow_up: |
| [backend/src/seed/factories/reports.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/672/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvc2VlZC9mYWN0b3JpZXMvcmVwb3J0cy5qcw==) | `0% <0%> (ø)` | :arrow_up: |
| [webapp/components/Editor/plugins/eventHandler.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/672/diff?src=pr&el=tree#diff-d2ViYXBwL2NvbXBvbmVudHMvRWRpdG9yL3BsdWdpbnMvZXZlbnRIYW5kbGVyLmpz) | `20% <0%> (ø)` | :arrow_up: |
| [backend/src/activitypub/routes/verify.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/672/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvYWN0aXZpdHlwdWIvcm91dGVzL3ZlcmlmeS5qcw==) | `0% <0%> (ø)` | :arrow_up: |
| [backend/src/activitypub/utils/actor.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/672/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvYWN0aXZpdHlwdWIvdXRpbHMvYWN0b3IuanM=) | `0% <0%> (ø)` | :arrow_up: |
| [backend/src/bootstrap/directives.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/672/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvYm9vdHN0cmFwL2RpcmVjdGl2ZXMuanM=) | `0% <0%> (ø)` | :arrow_up: |
| [...nd/src/middleware/notifications/extractMentions.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/672/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvbWlkZGxld2FyZS9ub3RpZmljYXRpb25zL2V4dHJhY3RNZW50aW9ucy5qcw==) | `100% <0%> (ø)` | :arrow_up: |
| ... and [48 more](https://codecov.io/gh/Human-Connection/Human-Connection/pull/672/diff?src=pr&el=tree-more) | |

File: webapp/pages/profile/_id/_slug.vue [hunk 1]
```diff
@@ -252,42 +259,62 @@
                     </no-ssr>
                   </ds-space>
                 </ds-flex-item>
-                <!--<ds-flex-item class="ds-tab-nav-item">
-                <ds-space margin="small">-->
-                <!-- TODO: find better solution for rendering errors -->
-                <!--
+                 
+                <ds-flex-item
+                  v-tooltip="{content: 'Your Comments', placement: 'right', delay: { show: 500 }}"
```
Comment (Tirokk) [inline webapp/pages/profile/_id/_slug.vue:264]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-26T10:42:50Z" title="Sunday, May 26th 2019, 12:42:50 pm +02:00">May 26, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -252,42 +259,62 @@
                     </no-ssr>
                   </ds-space>
                 </ds-flex-item>
-                <!--<ds-flex-item class="ds-tab-nav-item">
-                <ds-space margin="small">-->
-                <!-- TODO: find better solution for rendering errors -->
-                <!--
+                 
+                <ds-flex-item
+                  v-tooltip="{content: 'Your Comments', placement: 'right', delay: { show: 500 }}"
```


Same same :point_up:

File: webapp/pages/profile/_id/_slug.vue [hunk 2]
```diff
@@ -252,42 +259,62 @@
...
-                <!--<ds-flex-item class="ds-tab-nav-item">
-                <ds-space margin="small">-->
-                <!-- TODO: find better solution for rendering errors -->
-                <!--<no-ssr>
-                      <ds-number :label="$t('profile.shouted')">
-                        <hc-count-to slot="count" :end-val="user.shoutedCount"/>
+                </ds-flex-item>  
+                 
+                <ds-flex-item 
+                  v-tooltip="{content: 'Your Shouted', placement: 'left', delay: { show: 500 }}"
```
Comment (Tirokk) [inline webapp/pages/profile/_id/_slug.vue:283]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-26T10:43:23Z" title="Sunday, May 26th 2019, 12:43:23 pm +02:00">May 26, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -252,42 +259,62 @@
                     </no-ssr>
                   </ds-space>
                 </ds-flex-item>
-                <!--<ds-flex-item class="ds-tab-nav-item">
-                <ds-space margin="small">-->
-                <!-- TODO: find better solution for rendering errors -->
-                <!--
+                 
+                <ds-flex-item
+                  v-tooltip="{content: 'Your Comments', placement: 'right', delay: { show: 500 }}"
+                  class="ds-tab-nav-item "
+                  @click.prevent="tabActivity" >
+                  <ds-space margin="small">  
+                    <!--TODO: find better solution for rendering errors -->                
                     <no-ssr>
-                      <ds-number :label="$t('profile.commented')">
-                        <hc-count-to slot="count" :end-val="user.commentsCount"/>
+                    <ds-number :label="$t('common.comment', null, user.commentsCount)" 
+                      class="pointer">
+                        <hc-count-to
+                          slot="count"
+                          :end-val="user.commentsCount"
+                           
+                        />
                       </ds-number>
                     </no-ssr>
                   </ds-space>
-                </ds-flex-item>
-                -->
-                <!--<ds-flex-item class="ds-tab-nav-item">
-                <ds-space margin="small">-->
-                <!-- TODO: find better solution for rendering errors -->
-                <!--<no-ssr>
-                      <ds-number :label="$t('profile.shouted')">
-                        <hc-count-to slot="count" :end-val="user.shoutedCount"/>
+                </ds-flex-item>  
+                 
+                <ds-flex-item 
+                  v-tooltip="{content: 'Your Shouted', placement: 'left', delay: { show: 500 }}"
```


Same.

Comment (Tirokk) [inline webapp/pages/profile/_id/_slug.vue:283]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-26T10:43:23Z" title="Sunday, May 26th 2019, 12:43:23 pm +02:00">May 26, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -252,42 +259,62 @@
                     </no-ssr>
                   </ds-space>
                 </ds-flex-item>
-                <!--<ds-flex-item class="ds-tab-nav-item">
-                <ds-space margin="small">-->
-                <!-- TODO: find better solution for rendering errors -->
-                <!--
+                 
+                <ds-flex-item
+                  v-tooltip="{content: 'Your Comments', placement: 'right', delay: { show: 500 }}"
+                  class="ds-tab-nav-item "
+                  @click.prevent="tabActivity" >
+                  <ds-space margin="small">  
+                    <!--TODO: find better solution for rendering errors -->                
                     <no-ssr>
-                      <ds-number :label="$t('profile.commented')">
-                        <hc-count-to slot="count" :end-val="user.commentsCount"/>
+                    <ds-number :label="$t('common.comment', null, user.commentsCount)" 
+                      class="pointer">
+                        <hc-count-to
+                          slot="count"
+                          :end-val="user.commentsCount"
+                           
+                        />
                       </ds-number>
                     </no-ssr>
                   </ds-space>
-                </ds-flex-item>
-                -->
-                <!--<ds-flex-item class="ds-tab-nav-item">
-                <ds-space margin="small">-->
-                <!-- TODO: find better solution for rendering errors -->
-                <!--<no-ssr>
-                      <ds-number :label="$t('profile.shouted')">
-                        <hc-count-to slot="count" :end-val="user.shoutedCount"/>
+                </ds-flex-item>  
+                 
+                <ds-flex-item 
+                  v-tooltip="{content: 'Your Shouted', placement: 'left', delay: { show: 500 }}"
```


Same.

File: webapp/pages/profile/_id/_slug.vue [hunk 3]
```diff
@@ -252,42 +259,62 @@
...
             </ds-card>
           </ds-flex-item>
+
+      
+
           <ds-flex-item style="text-align: center">
             <ds-button
               v-if="myProfile"
-              v-tooltip="{content: 'Create a new Post', placement: 'left', delay: { show: 500 }}"
+              v-tooltip="{content: 'Create a new Post', placement: 'right', delay: { show: 500 }}"
```
Comment (Tirokk) [inline webapp/pages/profile/_id/_slug.vue:308]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-26T10:44:54Z" title="Sunday, May 26th 2019, 12:44:54 pm +02:00">May 26, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -252,42 +259,62 @@
                     </no-ssr>
                   </ds-space>
                 </ds-flex-item>
-                <!--<ds-flex-item class="ds-tab-nav-item">
-                <ds-space margin="small">-->
-                <!-- TODO: find better solution for rendering errors -->
-                <!--
+                 
+                <ds-flex-item
+                  v-tooltip="{content: 'Your Comments', placement: 'right', delay: { show: 500 }}"
+                  class="ds-tab-nav-item "
+                  @click.prevent="tabActivity" >
+                  <ds-space margin="small">  
+                    <!--TODO: find better solution for rendering errors -->                
                     <no-ssr>
-                      <ds-number :label="$t('profile.commented')">
-                        <hc-count-to slot="count" :end-val="user.commentsCount"/>
+                    <ds-number :label="$t('common.comment', null, user.commentsCount)" 
+                      class="pointer">
+                        <hc-count-to
+                          slot="count"
+                          :end-val="user.commentsCount"
+                           
+                        />
                       </ds-number>
                     </no-ssr>
                   </ds-space>
-                </ds-flex-item>
-                -->
-                <!--<ds-flex-item class="ds-tab-nav-item">
-                <ds-space margin="small">-->
-                <!-- TODO: find better solution for rendering errors -->
-                <!--<no-ssr>
-                      <ds-number :label="$t('profile.shouted')">
-                        <hc-count-to slot="count" :end-val="user.shoutedCount"/>
+                </ds-flex-item>  
+                 
+                <ds-flex-item 
+                  v-tooltip="{content: 'Your Shouted', placement: 'left', delay: { show: 500 }}"
+                  class="ds-tab-nav-item" >
+                  <ds-space margin="small">  
+                    <!-- TODO: find better solution for rendering errors -->
+                    <no-ssr>
+                      <ds-number :label="$t('common.shout', null, user.shoutedCount)" 
+                        class="pointer">
+                        <hc-count-to
+                          slot="count"
+                          :end-val="user.shoutedCount"
+                        />
                       </ds-number>
                     </no-ssr>
                   </ds-space>
-                </ds-flex-item>-->
+                </ds-flex-item>  
+
               </ds-flex>
             </ds-card>
           </ds-flex-item>
+
+      
+
           <ds-flex-item style="text-align: center">
             <ds-button
               v-if="myProfile"
-              v-tooltip="{content: 'Create a new Post', placement: 'left', delay: { show: 500 }}"
+              v-tooltip="{content: 'Create a new Post', placement: 'right', delay: { show: 500 }}"
```


..translations :gb: : :de:

File: webapp/pages/profile/_id/_slug.vue [hunk 4]
```diff
@@ -393,6 +422,16 @@ export default {
...
+    },
+     tabActivity(str, e) {
+      console.log("tabActivity");
+      console.log(str);
+      console.log(e);
+     // document.removeClass('ds-tab-nav-item-active');
+     // this.class('.ds-tab-nav-item-active')
+     // $$('.ds-tab-nav-item-active').removeClass('ds-tab-nav-item-active');
+     // $$(activTab).class("")
+     // $$(this).attr('class', 'ds-tab-nav-item-active');
```
Comment (Tirokk) [inline webapp/pages/profile/_id/_slug.vue:434]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-26T10:45:39Z" title="Sunday, May 26th 2019, 12:45:39 pm +02:00">May 26, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -393,6 +422,16 @@ export default {
     userName() {
       const { name } = this.user || {}
       return name || this.$t('profile.userAnonym')
+    },
+     tabActivity(str, e) {
+      console.log("tabActivity");
+      console.log(str);
+      console.log(e);
+     // document.removeClass('ds-tab-nav-item-active');
+     // this.class('.ds-tab-nav-item-active')
+     // $$('.ds-tab-nav-item-active').removeClass('ds-tab-nav-item-active');
+     // $$(activTab).class("")
+     // $$(this).attr('class', 'ds-tab-nav-item-active');
```


@ogerly this method does not do anything except `console.log`, please remove.

File: webapp/pages/profile/_id/_slug.vue [hunk 5]
```diff
@@ -239,11 +241,16 @@
           <ds-flex-item class="profile-top-navigation">
             <ds-card class="ds-tab-nav">
               <ds-flex>
-                <ds-flex-item class="ds-tab-nav-item ds-tab-nav-item-active">
+
+                <ds-flex-item
+                    v-tooltip="{content: 'Your Posts', placement: 'right', delay: { show: 500 }}"
```
[the remark this answers is not in the document]
Comment (Tirokk) [reply webapp/pages/profile/_id/_slug.vue:246]: > [ogerly](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-05-27T10:27:20Z" title="Monday, May 27th 2019, 12:27:20 pm +02:00">May 27, 2019</time>_

---


gif

**Comment by Tirokk** [comment]
> [ogerly](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-05-27T16:16:34Z" title="Monday, May 27th 2019, 6:16:34 pm +02:00">May 27, 2019</time>_

---


> Behebt diese PR auch das Problem, dass der entsprechende Filter für die Userposts verwendet wird? 
> -> Werden die vom Benutzer geschrienen / kommentierten Posts richtig angezeigt?

No, not yet. It's pure css. I'm trying my hand at it right now.

**Comment by Tirokk** [comment]
> [ogerly](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-06-05T08:58:26Z" title="Wednesday, June 5th 2019, 10:58:26 am +02:00">Jun 5, 2019</time>_

---


Sorry I'm still learning git!
this PR is closed.
continue here: https://github.com/Human-Connection/Human-Connection/pull/752

**Review by Mogge — CHANGES_REQUESTED** [review]
Looks good except the translations and the log output.

nice 

Because the PR adds CSS it would be super cool if you could include a screenshot of the result. It would give me a better impression how it looks like without the need of checking out the source code myself.

**Pull request closed without merging**